lib/eslint.js

```diff
@@ -617,33 +657,45 @@ module.exports = (function() {
             Object.keys(config.rules).filter(function(key) {
                 return getRuleSeverity(config.rules[key]) > 0;
             }).forEach(function(key) {
-
-                var ruleCreator = rules.get(key),
-                    severity = getRuleSeverity(config.rules[key]),
-                    options = getRuleOptions(config.rules[key]),
-                    rule;
-
-                if (ruleCreator) {
-                    try {
-                        rule = ruleCreator(new RuleContext(
-                            key, api, severity, options,
-                            config.settings, config.ecmaFeatures
-                        ));
-
-                        // add all the node types as listeners
-                        Object.keys(rule).forEach(function(nodeType) {
-                            api.on(nodeType, timing.enabled
-                                ? timing.time(key, rule[nodeType])
-                                : rule[nodeType]
-                            );
-                        });
-                    } catch(ex) {
-                        ex.message = "Error while loading rule '" + key + "': " + ex.message;
-                        throw ex;
+                var ruleCreator,
+                    severity,
+                    options,
+                    rule,
+                    deprecationMsg;
+
+                if (!rules.get(key)) {
```

Comment:
Really, this change is supposed to support two cases:

1. A previously deprecated rule has been removed and replaced with another rule. In that case, we need a message saying "foo rule was removed, please use bar rule instead.
1. A rule doesn't exist. In that case, we need a message saying, "Can't find foo rule.:

I don't think it's necessary to worry about anything other than these cases right now, so don't worry about plugins or deprecations where the rule hasn't been removed.